gubernator/github/handlers.py

```diff
@@ -0,0 +1,229 @@
+# Copyright 2016 The Kubernetes Authors All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import classifier
+import cgi
+import datetime
+import hashlib
+import hmac
+import logging
+import json
+import zlib
+
+import webapp2
+
+from google.appengine.api import taskqueue
+from google.appengine.ext import deferred, ndb
+from google.appengine.datastore.datastore_query import Cursor
+
+import models
+
+
+try:
+    WEBHOOK_SECRET = open('webhook_secret').read().strip()
+except IOError:
+    logging.warning('unable to load webhook secret')
+    WEBHOOK_SECRET = 'default'
+
+def secure_equal(a, b):
+    '''
+    Compare a and b without leaking timing information.
+
+    May reveal length of b.
+    '''
+    if len(a) != len(b):
+        return False
+    return sum(ord(a[i]) ^ ord(b[i]) for i in xrange(len(a))) == 0
+
+
+def make_signature(body):
+    hmac_instance = hmac.HMAC(WEBHOOK_SECRET, body, hashlib.sha1)
+    return 'sha1=' + hmac_instance.hexdigest()
+
+
+class GithubHandler(webapp2.RequestHandler):
```

Comment:
I would appreciate comments for just about every class in this file. The names are pretty good but there's still some tricky stuff there.